projects/bp-gallery/cypress/e2e/facetags.cy.ts

```diff
@@ -0,0 +1,105 @@
+import { login, logout } from '../utils/login-utils';
+
+describe('face tagging', () => {
+  before(() => {
+    cy.visit('/');
+    login();
+    cy.visit('/picture/1');
+  });
+
+  after(() => {
+    cy.visit('/');
+    cy.get('.nav-bar').contains('Mehr...').click();
+    cy.get('.MuiPaper-root').contains('Personen').click();
+    cy.contains('.MuiDataGrid-row', 'TestPerson').find('[data-testid="DeleteIcon"]').click();
+    cy.get('.MuiButton-root').contains('Bestätigen').click();
+    cy.contains('.MuiDataGrid-row', 'TestPerson').should('not.exist');
+    cy.contains('.MuiDataGrid-row', 'Personentest').find('[data-testid="DeleteIcon"]').click();
+    cy.get('.MuiButton-root').contains('Bestätigen').click();
+    cy.contains('.MuiDataGrid-row', 'Personentest').should('not.exist');
+    cy.visit('/');
+    logout();
+  });
+
+  it('create face tags', () => {
+    cy.get('.facetag').should('not.exist');
+    cy.contains('Personen markieren').should('not.exist');
+    cy.contains('Personenmarkierungen ausblenden').should('not.exist');
+    cy.contains('Personenmarkierungen anzeigen').should('not.exist');
+
+    cy.contains('.field-content', 'Personen').find('input').click();
+    cy.contains('.field-content', 'Personen').find('input').clear();
+    cy.contains('.field-content', 'Personen').find('input').type('TestPerson');
+    cy.contains('TestPerson hinzufügen').click();
+    cy.contains('.save-state', 'Gespeichert');
+
+    cy.contains('.field-content', 'Personen').find('input').click();
+    cy.contains('.field-content', 'Personen').find('input').clear();
+    cy.contains('.field-content', 'Personen').find('input').type('Personentest');
+    cy.contains('Personentest hinzufügen').click();
+    cy.contains('.save-state', 'Gespeichert');
+
+    cy.contains('Personen markieren').click();
+    cy.contains('Personen bearbeiten');
+
+    cy.contains('TestPerson').click();
+    cy.get('.picture').click('left');
+    cy.contains('.facetag', 'TestPerson').find('[data-testid="CancelIcon"]');
+
+    cy.contains('Personentest').click();
+    cy.get('.picture').click();
+    cy.contains('.facetag', 'Personentest').find('[data-testid="CancelIcon"]');
+  });
+
+  it('hide and display facetags', () => {
+    cy.contains('Personen bearbeiten').click();
+    cy.contains('Personenmarkierungen anzeigen').should('not.exist');
+    cy.contains('Personenmarkierungen ausblenden').click();
+    cy.get('.facetag').should('not.exist');
+    cy.contains('Personenmarkierungen ausblenden').should('not.exist');
+    cy.contains('Personenmarkierungen anzeigen').click();
+    cy.get('.facetag').should('have.length', 2);
+    cy.contains('Personenmarkierungen ausblenden');
+
+    //test wether it works as a regular user too
+    cy.visit('/');
+    logout();
+    cy.visit('/picture/1');
+
```

Comment:
kann man sowas in eine subroutine auslagern?

Reply:
`login` macht absichtlich kein `cy.visit`, damit man das an der call-site machen kann und wenn gewünscht weglassen kann (um nicht den aktuellen Zustand zu verlieren). Man könnte einen Helper bauen, der `visit` und `login` nacheinander aufruft, aber ich finde es so auch in Ordnung.